src/libs/portfolio/getOnchainBalances.ts

```diff
@@ -176,14 +176,23 @@ export async function getNFTs(
       : null
 
     const token = mapToken(beforeToken)
+    const [receiving, sending]: [bigint[], bigint[]] = [[], []]
+    beforeToken[2].forEach((oldCollectible: bigint) => {
+      if (simulationToken?.collectibles && !simulationToken?.collectibles?.includes(oldCollectible))
+        sending.push(oldCollectible)
+    })
+    simulationToken?.collectibles?.forEach((newCollectible: bigint) => {
+      if (!beforeToken[2].includes(newCollectible)) receiving.push(newCollectible)
+    })
 
     return [
       beforeToken.error,
       {
         ...token,
         // Please refer to getTokens() for more info regarding `amountBeforeSimulation` calc
         simulationAmount: simulationToken ? simulationToken.amount - token.amount : undefined,
-        amountPostSimulation: simulationToken ? simulationToken.amount : token.amount
+        amountPostSimulation: simulationToken ? simulationToken.amount : token.amount,
```

Comment:
simulationAmount and amountPostSimulation are now redundant

Reply:
we should only rely only on `sending` and `receinving` for any changes to the balance in regards to NFTs in the context of executing an account op.

if we want to get the static *`'amount'`* for the collection we could refer to `.collectibles.length` anyway

if we take this approach the collectibles array should be from before the simulation

Reply:
wyt @Ivshti @jordan-enev

Reply:
Yes, the simulationAmount and amountPostSimulation fields represent a numeric/static change in the collectibles' length. The properties that Yosif introduced, sending and received, are much more informative. In terms of NFTs, both simulationAmount and amountPostSimulation seem irrelevant to me.